docs/snippets/ov_model_snippets.cpp

```diff
@@ -102,10 +102,8 @@ void model_inputs() {
 ov::Core core;
 std::shared_ptr<ov::Model> model = core.read_model("model.xml");
 //! [all_inputs_ouputs]
-/* Take information about all topology inputs */
-auto inputs = model->inputs();
-/* Take information about all topology outputs */
-auto outputs = model->outputs();
+auto ov_model_input = ov_model->input(original_fw_in_tensor_name);
+auto ov_model_output = ov_model->output(original_fw_out_tensor_name);
```

Comment:
@itikhono, please help @msmykx-intel to provide more explicit information about missing names, multiple names and a warning about `get_any_name`. We do not have enough information about names -- users writes incorrect code because part of the API is not demonstrated and users incorrectly interpret that part when discover it by their own.

Reply:
I believe we need to include 4 points:
1. User can use these methods to get inputs/outputs using indexes:

c++ 

auto ov_model_input = ov_model->input(index);
auto ov_model_output = ov_model->output(index);

py:

ov_model_input = model.input(index)
ov_model_output = model.output(index)

2. Using original framework tensor names
 ```
 c++:

auto ov_model_input = ov_model->input(original_fw_in_tensor_name);
auto ov_model_output = ov_model->output(original_fw_out_tensor_name);

 py:

ov_model_input = model.input(original_fw_in_tensor_name)
ov_model_output = model.output(original_fw_out_tensor_name)
```

3. User can get list of all tensor names via 

```
c++ 

for (const auto& input : model->inputs()) {
   auto tensor_names =  input.get_names(); // returns a list of the tensor names associated with this concrete tensor. This list may contain more names than it was in the original model but for inputs/outputs (Parameters/Resusts) it's guaranteed that all the names from original fw will be in the list.
   ... // user logic here
}

py:

    for ov_input in model.inputs:
        ov_tensor_names = ov_input.get_names()
        # user logic here
```

4. Additional methods to work with tensor/tensor_names

c++:

auto tensor = output->get_tensor(); //  User can get access to the tensor object
auto name = tensor.get_any_name(); // This method will return some tensor name from the tensor_names list

py: 

tensor = output.get_tensor()
name = tensor.get_any_name()

Reply:
We need to provide an explicit warning that names are optional. Just presenting a list of example is not enough. Example: outputs in PyTroch model may not have names. Use explicit names and indices instead of `index`, `original_fw_in_tensor_name` and so on.

Reply:
`... // user logic here` gives nothing.

Reply:
Avoid using examples with `get_any_name`. We actually need to minimize usage of that. Just mention that this method exists and it can return a list of names that can be empty.

Reply:
https://github.com/openvinotoolkit/openvino/pull/23482/commits/0e9ad71e1550db01a99239a570feddcd9a2e7bdd
https://github.com/openvinotoolkit/openvino/pull/23482/commits/79484f316f3e2d06575fa239ba6cdc7621e714f2

I used the code snippet with `index`, and I added the warning note that describes what @slyalin mentioned above about using names and `get_names`.